src/main/java/org/opensearch/neuralsearch/processor/TextImageEmbeddingProcessor.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.env.Environment;
+import org.opensearch.index.mapper.MapperService;
+import org.opensearch.ingest.AbstractProcessor;
+import org.opensearch.ingest.IngestDocument;
+import org.opensearch.neuralsearch.ml.MLCommonsClientAccessor;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * This processor is used for user input data text and image embedding processing, model_id can be used to indicate which model user use,
+ * and field_map can be used to indicate which fields needs embedding and the corresponding keys for the embedding results.
+ */
+@Log4j2
+public class TextImageEmbeddingProcessor extends AbstractProcessor {
+
+    public static final String TYPE = "text_image_embedding";
+    public static final String MODEL_ID_FIELD = "model_id";
+    public static final String EMBEDDING_FIELD = "embedding";
+    public static final String FIELD_MAP_FIELD = "field_map";
+    public static final String TEXT_FIELD_NAME = "text";
+    public static final String IMAGE_FIELD_NAME = "image";
+    public static final String INPUT_TEXT = "inputText";
+    public static final String INPUT_IMAGE = "inputImage";
+
+    @VisibleForTesting
+    private final String modelId;
+    private final String embedding;
+    private final Map<String, String> fieldMap;
+
+    private final MLCommonsClientAccessor mlCommonsClientAccessor;
+    private final Environment environment;
+
+    public TextImageEmbeddingProcessor(
+        String tag,
+        String description,
+        String modelId,
+        String embedding,
+        Map<String, String> fieldMap,
+        MLCommonsClientAccessor clientAccessor,
+        Environment environment
+    ) {
+        super(tag, description);
+        if (StringUtils.isBlank(modelId)) throw new IllegalArgumentException("model_id is null or empty, can not process it");
+        validateEmbeddingConfiguration(fieldMap);
+
+        this.modelId = modelId;
+        this.embedding = embedding;
+        this.fieldMap = fieldMap;
+        this.mlCommonsClientAccessor = clientAccessor;
+        this.environment = environment;
+    }
+
+    private void validateEmbeddingConfiguration(Map<String, String> fieldMap) {
+        if (fieldMap == null
+            || fieldMap.isEmpty()
+            || fieldMap.entrySet()
+                .stream()
+                .anyMatch(x -> StringUtils.isBlank(x.getKey()) || Objects.isNull(x.getValue()) || StringUtils.isBlank(x.getValue()))) {
```

Comment:
`isBlank` should handle null case
```suggestion
                .anyMatch(x -> StringUtils.isBlank(x.getKey()) || StringUtils.isBlank(x.getValue()))) {
```

Reply:
ack